File: frontend/src/types/company.ts
export interface Company {
    id: number;
    companyName: string;
    address: string;
    country: string;
    nameOfOwner: string;
    registrationNumber: string;
    role: string;
    category?: string;
    deliveryMethod?: string;
}

export interface PagedCompanies {
    content: Company[];
    totalPages: number;
    totalElements: number;
    number: number;
    size: number;
}

export interface CompanyCreateRequest {
    companyName: string;
    address: string;
    country: string;
    nameOfOwner: string;
    role: string;

    registrationNumber?: string;
    partnerDate?: string;
    category?: string;
    deliveryMethod?: string;
    logoPath?: string;
    signaturePath?: string;
}

File: src/main/java/com/tradeexport/backend/company/Company.java
package com.tradeexport.backend.company;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;
import lombok.NoArgsConstructor;
import java.time.LocalDateTime;

@Entity
@Getter
@Setter
@NoArgsConstructor
public class Company {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String companyName;
    private String address;
    private String country;
    private String registrationNumber;
    private String nameOfOwner;
    private LocalDateTime partnerDate;
    private String category;
    private String deliveryMethod;
    private String role;
    private String logoPath;
    private String signaturePath;
}
